matchms/filtering/reduce_to_number_of_peaks.py

```diff
@@ -19,14 +19,18 @@ def reduce_to_number_of_peaks(spectrum_in: SpectrumType, n_required: int = 1, n_
         Maximum number of peaks. Remove peaks if more peaks are found.
     ratio_desired:
         Set desired ratio between maximum number of peaks and parent mass.
+        For spectra without parent mass (e.g. GCMS spectra) this will raise an
+        error when ratio_desired is used.
         Default is None.
     """
     def _set_maximum_number_of_peaks_to_keep():
         parent_mass = spectrum.get("parent_mass", None)
         if parent_mass and ratio_desired:
             n_desired_by_mass = int(ceil(ratio_desired * parent_mass))
             return max(n_required, n_desired_by_mass)
-        return n_max
+        if not ratio_desired:
+            return n_max
+        raise ValueError("Cannot use ratio_desired for spectrum without parent_mass.")
```

Comment:
Can you add a test for when ratio_desired=None and spectrum has no parent.

Reply:
Added a test case for ratio_desired=None but with parent mass (the one without both is already present).